### ISSUE TITLE
Panics when a file name contains multi-byte characters and whitespaces.

### ISSUE THREAD

**Issue by baehyunsol** [nushell/nushell#9320]
### Describe the bug

It panics for no reason. I'll explain how to reproduce this panic.

### How to reproduce

1. Make a new directory.
2. Create files whose name consists of only whitespaces and multi-byte characters. See figure 1 below.
3. Type `` mv  `실 험 용` `` and place your cursor between `mv` and `` `실 험 용` ``. There are 2 whitespaces between `mv` and `` `실 험 용` ``.  It should be `` `실 험 용` ``, not `` `실 험 용 2` ``
4. Press tab
5. Press enter
6. It panics. See figure 2 below.

### Expected behavior

It must not panic.

### Screenshots

1. ![20230530225323](https://github.com/nushell/nushell/assets/47506131/5da76f5d-9aba-466c-8b31-7e066fdac6e0)
2. ![20230530225848](https://github.com/nushell/nushell/assets/47506131/e8643ce9-af22-472d-a362-e74f4bf1002e)

### Configuration

| key                | value                               |
| ------------------ | ----------------------------------- |
| version            | 0.80.0                              |
| branch             |                                     |
| commit_hash        |                                     |
| build_os           | linux-x86_64                        |
| build_target       | x86_64-unknown-linux-gnu            |
| rust_version       | rustc 1.69.0 (84c898d65 2023-04-16) |
| rust_channel       | stable-x86_64-unknown-linux-gnu     |
| cargo_version      | cargo 1.69.0 (6e9a83356 2023-04-12) |
| build_time         | 2023-05-27 19:28:20 +09:00          |
| build_rust_channel | release                             |
| features           | default, zip                        |
| installed_plugins  |                                     |

**Comment by baehyunsol** [comment]
When I press tab at `` mv   `실 험 용` ``, it becomse `` mv `실 험 용 `실 험 용` ``, because the names of both files starts with `실 험 용`. But it doesn't include closing `` ` ``.

I'm not sure whether it's the cause of the problem, but I guess it also has to be fixed.

**Comment by fdncred** [comment]
good job figuring it out. nice research! are you able to create a fix for it?

**Comment by baehyunsol** [comment]
> good job figuring it out. nice research! are you able to create a fix for it?

I'll give it a try!

**Comment by baehyunsol** [comment]
When tracking the stack backtrace, I found that [this](https://github.com/nushell/nushell/blob/a5dd93d1543da1b5ea47b93591c41a8997b20bb4/crates/nu-cli/src/syntax_highlight.rs#L17) function is called with these arguments: 

`` highlight(line: "mv `실 험 용 2` 험 용`", _cursor: 13) ``

There are 2 problems with that.

1. The `line` argument should be `` "mv `실 험 용 2` `실 험 용`" ``. I have no idea why a few characters are missing. They just vanished.
2. `_cursor` is wrong regardless the correctness of `line`. The index 12 is at the middle of `용` character. Each korean character takes 3 bytes, so the `용` is 12..15.

But `line` and `_cursor` are managed by another crate, reedline... Now I have another crate to debug

**Comment by fdncred** [comment]
I figured this was always about switching from bytes, chars(), or graphemes?

**Comment by baehyunsol** [comment]
Yeah it fails to calc the boundary of graphemes

**Comment by KAAtheWiseGit** [comment]
Yeah.  I tried working on the highlighting, but lacked the expertise, it's quite complex.
